### PR TITLE
cz/improvements in vcf_transform

### PR DESCRIPTION
**Pull request by ChuanyiZ**
- support Moss format VCF file
- support VCF files containing multiple samples
- finished todo: automatically extract sample name
- change default major and minor copy number to 1

### CODE REVIEW

File: vcf_transform/code/py_code/mutation.py [hunk 1]
```diff
@@ -28,13 +28,12 @@ def from_mutect_vcf_record(sample_id:str, vcf_record:vcf.model._Record):
 
         mutation_id = Mutation._construct_mutation_id(vcf_record)
 
-        #A25 -> AD 
-        #TODO: should these be parameters, or are they hardcoded for 'mutect'?
-        tumor_call = vcf_record.genotype('A25')
+        tumor_sample = vcf_reader.metadata["tumor_sample"]
+        tumor_call = vcf_record.genotype(tumor_sample)
```
Comment (pgroves) [inline vcf_transform/code/py_code/mutation.py:32]: Much better. Thank you.

File: vcf_transform/code/py_code/main.py [hunk 2]
```diff
@@ -11,8 +11,8 @@ def main(args):
     print("main.py: got the args: " + str(args))
 
     vcf_type:str = args[0]
-    if not vcf_type == 'mutect':
-        raise Exception('Can only understand vcf from mutect')
+    if not vcf_type == 'mutect' and vcf_type != 'moss':
```
Comment (pgroves) [inline vcf_transform/code/py_code/main.py:14]: Could you please write this like:
`not vcf_type in ['mutect', 'moss']`

**Review by pgroves — APPROVED** [review]
I just had one minor styling comment but otherwise this looks good. You can merge this when you think it's ready.

File: vcf_transform/code/py_code/main.py (hunk 2 above)
Comment (ChuanyiZ) [reply vcf_transform/code/py_code/main.py:14]: Sure

Comment (matthewberry) [reply vcf_transform/code/py_code/main.py:14]: or, alternatively, `vcf_type not in ['mutect', 'moss']` (moving the `not`, which some people consider more readable and "pythonic")

File: vcf_transform/code/py_code/mutation.py [hunk 3]
```diff
@@ -44,6 +43,61 @@ def from_mutect_vcf_record(sample_id:str, vcf_record:vcf.model._Record):
...
+        tumor_samples = vcf_reader.metadata["tumor_sample"]
+
+        for rec in vcf_reader:
+            if len(rec.FILTER) == 0:
+                for sample in tumor_samples:
+                    mutation_id = Mutation._construct_mutation_id(rec)
+                    tumor_call = rec.genotype(sample)
+                    call_data = tumor_call.data
+                    DP = int(call_data.DP)
+                    TCOUNT = int(call_data.TCOUNT)
```
Comment (matthewberry) [inline vcf_transform/code/py_code/mutation.py:89]: just a minor style thing, but the `DP` and `TCOUNT` variables are local to this block, right? in that case, lowercase variable names would signal to anyone reading the code that the variables aren't shared with the rest of the module.

**Review by matthewberry — APPROVED** [review]

File: vcf_transform/code/py_code/mutation.py (hunk 3 above)
Comment (ChuanyiZ) [reply vcf_transform/code/py_code/mutation.py:89]: Yes, that's a good point.

**Pull request merged**